tests/console/sshd.pm

```diff
@@ -8,57 +8,77 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# Summary: Very old test to verify sshd starts up.
+# Summary: Test to verify sshd starts and accepts connections.
 #  We need this test to succeed for followup tests using ssh localhost
-# Maintainer: Stephan Kulow <coolo@suse.de>
+#  This regression test has also an interactive part (in VirtIO console)
+# Maintainer: Stephan Kulow <coolo@suse.de>, Pavel Dostál <pdostal@suse.cz>
 
+use warnings;
 use base "consoletest";
 use strict;
 use testapi;
-use utils;
+use utils qw(systemctl exec_and_insert_password);
 use version_utils qw(is_virtualization_server is_upgrade);
 
-# check if sshd works
 sub run {
     my $self = shift;
     # new user to test sshd
     my $ssh_testman        = "sshboy";
-    my $ssh_testman_passwd = "let3me2in1";
+    my $ssh_testman_passwd = $testapi::password;
 
     select_console 'root-console';
 
+    # Stop the firewall if it's available
     if (is_upgrade && check_var('ORIGIN_SYSTEM_VERSION', '11-SP4')) {
         record_info("SuSEfirewall2 not available", "bsc#1090178: SuSEfirewall2 service is not available after upgrade from SLES11 SP4 to SLES15");
     }
     else {
         systemctl 'stop ' . $self->firewall if !is_virtualization_server;
     }
-    script_run('chkconfig sshd on');
-    assert_script_run("chkconfig sshd on", 60);
-    assert_script_run("rcsshd restart",    60);    # will do nothing if it is already running
 
-    sleep 3;                                       # give the daemon some time to start
+    # Restart sshd and check it's status
+    systemctl 'restart sshd';
+    systemctl 'status sshd';
 
-    script_run('rcsshd status', 0);
-    assert_screen 'test-sshd-1';
+    # Check that the daemons listens on right addresses/ports
+    assert_script_run q(ss -pnl4 | egrep 'tcp.*LISTEN.*:22.*sshd');
+    assert_script_run q(ss -pnl6 | egrep 'tcp.*LISTEN.*:22.*sshd');
```

Comment:
Why do we need this? We check if we can connect later one anyway, right? Or is it to check that the daemon listens not just on localhost?

Reply:
Yes, I check it listens everywhere (IPv4 as well as v6).